src/components/App.tsx

```diff
@@ -90,15 +86,21 @@ class App extends React.Component<RouteProps, State> {
         this.setState({
           isLoading: false,
           networkErrorOccurred: true,
-          errorMessage
+          errorMessage,
+          logEntries: []
         });
       });
-  }
 
-  private handleAddEntry = (entry: LiftLogEntry) =>
-    this.liftLogService
-      .addEntry(this.logName, entry)
-      .then(() => this.reloadLifts());
+  private async handleAddEntry(entry: LiftLogEntry) {
+    this.setState(prevState => ({
+      isLoading: true,
+      loadingMessage: `Adding entry for ${entry.name}...`,
+      logEntries: prevState.logEntries
+    }));
+
+    await this.liftLogService.addEntry(this.logName, entry);
+    await this.reloadLifts();
```

Comment:
`reloadLifts` is not marked `async`, I guess you remove `await` from here then?

Reply:
`await` is applied to an object of type `Promise`, not an "async function".
If I understand async/await correctly (I think it's pretty similar to C#), then the `async` keyword only affects what is allowed inside the method marked as `async`. Namely the reason to mark a function `async` is to be able to do `awaits` inside the function body.